runtime/src/transaction_batch.rs

```diff
@@ -51,7 +51,14 @@ impl<'a, 'b> TransactionBatch<'a, 'b> {
 // Unlock all locked accounts in destructor.
 impl<'a, 'b> Drop for TransactionBatch<'a, 'b> {
     fn drop(&mut self) {
-        self.bank.unlock_accounts(self)
+        if self.needs_unlock() {
+            self.set_needs_unlock(false);
```

Comment:
I realize we had this previously, but I'm struggling to think why this is necessary. We're in middle of dropping this batch, so it won't get dropped and attempt unlock again, right?

Reply:
Yup, it's really not necessary to set to `false` here